### PR TITLE
No response for some more IWAP Commands

### PR DESCRIPTION
**Pull request by Nikutrax**
No response for following IWAP Commands:
AP33
AP34
AP84
AP85

Attached is the protocol where the changes are visible. On page 20, you can see AP33; on page 21, AP34; and on page 23, AP84/AP85.
[Gps Watch protocol_2023.pdf](https://github.com/user-attachments/files/17185670/Gps.Watch.protocol_2023.pdf)

### CODE REVIEW

File: src/main/java/org/traccar/protocol/TrvProtocolDecoder.java [hunk 1]
```diff
@@ -181,6 +181,8 @@ protected Object decode(
             } else if (type.equals("AP14") && !id.equals("IW")) {
                 channel.writeAndFlush(new NetworkMessage(responseHeader + ",0.000,0.000#", remoteAddress));
             } else if (!type.equals("AP12") && !type.equals("AP14")
+                    && !type.equals("AP33") && !type.equals("AP34")
+                    && !type.equals("AP84") && !type.equals("AP85")
```
Comment (tananaev) [inline src/main/java/org/traccar/protocol/TrvProtocolDecoder.java:185]: I think we should do something like `!Set.of("AP12", ..., "AP85").includes(type)`.

Comment (Nikutrax) [reply src/main/java/org/traccar/protocol/TrvProtocolDecoder.java:185]: if (channel != null) {
    String responseHeader = id + (char) (type.charAt(0) + 1) + type.substring(1);
    
    Set<String> excludedTypes = new HashSet<>(Arrays.asList("AP12", "AP14", "AP33", "AP34", "AP84", "AP85"));

    if (type.equals("AP00") && id.equals("IW")) {
        // Zeitformatierung nur einmal erstellen
        String time = new SimpleDateFormat("yyyyMMddHHmmss").format(new Date());
        channel.writeAndFlush(new NetworkMessage(responseHeader + "," + time + ",0#", remoteAddress));
    } else if (type.equals("AP14") && !id.equals("IW")) {
        channel.writeAndFlush(new NetworkMessage(responseHeader + ",0.000,0.000#", remoteAddress));
    } else if (!excludedTypes.contains(type) && !sentence.substring(responseHeader.length() + 1).matches("^\\d{6}$")) {
        channel.writeAndFlush(new NetworkMessage(responseHeader + "#", remoteAddress));
    }
}

**Comment by Nikutrax** [comment]
like this?

File: src/main/java/org/traccar/protocol/TrvProtocolDecoder.java (hunk 1 above)
Comment (tananaev) [reply src/main/java/org/traccar/protocol/TrvProtocolDecoder.java:185]: Why can't you use `Set.of` inline, like I suggested?

**Comment by Nikutrax** [comment]
like this ?

if (channel != null) {
    String responseHeader = id + (char) (type.charAt(0) + 1) + type.substring(1);
    
    // Unveränderliche Menge der AP-Typen, die ausgeschlossen werden sollen
    Set<String> excludedTypes = Set.of("AP12", "AP14", "AP33", "AP34", "AP84", "AP85");

    if (type.equals("AP00") && id.equals("IW")) {
        // Zeitformatierung nur einmal erstellen
        String time = new SimpleDateFormat("yyyyMMddHHmmss").format(new Date());
        channel.writeAndFlush(new NetworkMessage(responseHeader + "," + time + ",0#", remoteAddress));
    } else if (type.equals("AP14") && !id.equals("IW")) {
        channel.writeAndFlush(new NetworkMessage(responseHeader + ",0.000,0.000#", remoteAddress));
    } else if (!excludedTypes.contains(type) && !sentence.substring(responseHeader.length() + 1).matches("^\\d{6}$")) {
        channel.writeAndFlush(new NetworkMessage(responseHeader + "#", remoteAddress));
    }
}

**Comment by Nikutrax** [comment]
so better?

**Comment by tananaev** [comment]
No, it seems like you ignored my last comment.

**Comment by Nikutrax** [comment]
i don´t ignore, but i don´t understand what you mean

**Comment by Nikutrax** [comment]
if (channel != null) {
    String responseHeader = id + (char) (type.charAt(0) + 1) + type.substring(1);

   if (type.equals("AP00") && id.equals("IW")) {
        String time = new SimpleDateFormat("yyyyMMddHHmmss").format(new Date());
        channel.writeAndFlush(new NetworkMessage(responseHeader + "," + time + ",0#", remoteAddress));
    } else if (type.equals("AP14") && !id.equals("IW")) {
        channel.writeAndFlush(new NetworkMessage(responseHeader + ",0.000,0.000#", remoteAddress));
    } else if (!Set.of("AP12", "AP14", "AP33", "AP34", "AP84", "AP85").contains(type)
            && !sentence.substring(responseHeader.length() + 1).matches("^\\d{6}$")) {
        channel.writeAndFlush(new NetworkMessage(responseHeader + "#", remoteAddress));
    }
}

**Comment by tananaev** [comment]
Make sure you test your changes first before pushing here.

**Comment by Nikutrax** [comment]
else if (!Set.of("AP12", "AP14", "AP33", "AP34", "AP84", "AP85").contains(type)

**Comment by Nikutrax** [comment]
if (channel != null) {
    String responseHeader = id + (char) (type.charAt(0) + 1) + type.substring(1);
    
    if (type.equals("AP00") && id.equals("IW")) {
        String time = new SimpleDateFormat("yyyyMMddHHmmss").format(new Date());
        channel.writeAndFlush(new NetworkMessage(responseHeader + "," + time + ",0#", remoteAddress));
    } else if (type.equals("AP14") && !id.equals("IW")) {
        channel.writeAndFlush(new NetworkMessage(responseHeader + ",0.000,0.000#", remoteAddress));
    } else if (!Set.of("AP12", "AP14", "AP33", "AP34", "AP84", "AP85").contains(type) 
               && !sentence.substring(responseHeader.length() + 1).matches("^\\d{6}$")) {
        channel.writeAndFlush(new NetworkMessage(responseHeader + "#", remoteAddress));
    }
}

**Comment by Nikutrax** [comment]
do you know the misstake?

**Comment by tananaev** [comment]
I'm going to close this for now. Please make the changes locally, test then, run all the checks and then submit the PR.

**Pull request closed without merging**